hw/top_earlgrey/ip/rstmgr/system_rstmgr_pkg.core

```diff
@@ -3,14 +3,14 @@ CAPI=2:
 # Licensed under the Apache License, Version 2.0, see LICENSE for details.
 # SPDX-License-Identifier: Apache-2.0
 name: "lowrisc:systems:rstmgr_pkg:0.1"
-description: "Top level Reset Manager Package"
+description: "Auto-generated reset manager package for top_earlgrey"
 
 filesets:
   files_rtl:
     depend:
       - lowrisc:ip:pwrmgr_pkg
-      - lowrisc:ip:rstmgr_reg
-      - lowrisc:ip:alert_handler_reg
+      - lowrisc:systems:rstmgr_reg
```

Comment:
for example this part... if you imagine a non templated module.. let's say uart for some reason has to reference rstmgr, do we really want it to reference the system area? or the ip version? My thinking was that everyone should point to the ip version, and from there we re-direct based on whether it's ip or top. 

let me know what you think.

Reply:
That's a good example. I think we always want it to point to the IP version (and NOT the system area, except we are in a system IP as it is the case here). And this also possible with the new cores files. The downside is you have to depend on the whole IP and not just the registers or package core.

Reply:
so just to make sure i follow.. 
are you saying, for argument's sake, if `uart` needed a parameter from `rstmgr`, instead of putting `rstmgr_pkg.core` in its dependency, in this new scheme it would need to put `rstmgr.core`? 

Does that imply then that we should just get rid of `rstmgr_pkg.core`? Because there wouldn't really be a case you can reference that independently right..? Especially for `rstmgr / clkmgr / pwrmgr`, these trio of blocks seem to tend to reference one another a lot.

I need to digest that a bit.. but i THINK i'm okay with it.  It's just up to now I've been consciously separating the "package" files from the "design" files, but not for any real reason other than seeing other modules do it also. 

Do you think it makes sense to bring this topic up quickly at the silicon meeting?  Ultimately I think the one that makes the most sense is the one that is most readable / understandable by everyone at first glance.  And it may be that what you're showing here is the way to go.  It would just be good to make sure everyone is okay with it.

Reply:
> are you saying, for argument's sake, if `uart` needed a parameter from `rstmgr`, instead of putting `rstmgr_pkg.core` in its dependency, in this new scheme it would need to put `rstmgr.core`?

Yes, if that is working. In my opinion, the simpler the better. But there might be cases where that's not sufficient.

> Does that imply then that we should just get rid of `rstmgr_pkg.core`? Because there wouldn't really be a case you can reference that independently right..? Especially for `rstmgr / clkmgr / pwrmgr`, these trio of blocks seem to tend to reference one another a lot.

I didn't think that far but yes that's right. There is little use of the `*_pkg/reg_pkg.core` files if we can't reference them directly. Having 3 separate core files for these 3 IPs gives us 18 core files (9 in the IP and 9 in the system domain). That's quite a lot if something doesn't work or we need to change something, and this is what led me to do this "cleanup". I think the fewer core files the better.

> Do you think it makes sense to bring this topic up quickly at the silicon meeting? Ultimately I think the one that makes the most sense is the one that is most readable / understandable by everyone at first glance. And it may be that what you're showing here is the way to go. It would just be good to make sure everyone is okay with it.

I guess there won't be a silicon meeting next week. But I would be curious to get also @imphil's feedback on this matter. It might be that with a single core file per IP we will run into other problems later. Philipp is usually very good at recognizing such things.

Reply:
For the record: We have decided to leave this as it is, i.e., to fan out inside `rstmgr`, `rstmgr_pgk` and `rstmgr_reg`. The main rationale behind this is that for DV, we don't want to pull in too much code but just the headers.